eth/src/main/java/net/consensys/cava/eth/Transaction.java

```diff
@@ -104,14 +104,15 @@ public static Transaction readFrom(RLPReader reader) {
       throw new RLPException("Additional bytes present at the end of the encoding");
     }
 
-    Byte v = null;
+    int encodedVasInt = encodedV.toInt();
```

Comment:
Probably want to check that it is <= 4 bytes before calling here so that you can throw an RLPException rather than a IllegalStateException

Reply:
Indeed. OK I'll change the logic so the RLPReader reads an int directly and will send errors right away. Some of the tests failed because of this.